crates/bevy_geometry/src/lib.rs

```diff
@@ -0,0 +1,383 @@
+use bevy_math::*;
+use bevy_reflect::Reflect;
+use std::error::Error;
+use std::fmt;
+
+pub trait Primitive3d {
+    /// Returns true if this primitive is entirely on the outside (in the normal direction) of the
+    /// supplied plane.
+    fn outside_plane(&self, plane: Plane) -> bool;
+}
+
+#[derive(Debug, Clone)]
+pub enum PrimitiveError {
+    MinGreaterThanMax,
+    NonPositiveExtents,
+}
+impl Error for PrimitiveError {}
```

Comment:
instead of implementing `Error` and `Display` for this type.
You can use the `thiserror` crate. (we use it in other places of the bevy code base)
```rust
#[derive(Error, Debug, Clone)]
pub enum PrimitiveError {
    #[error("this is the error message")]
    MinGreaterThanMax,
    #[error("this is another error message")]
    NonPositiveExtents,
}
```